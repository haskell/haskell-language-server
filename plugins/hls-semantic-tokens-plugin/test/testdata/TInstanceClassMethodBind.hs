module TInstanceClassMethodBind where


data Foo = Foo Int
instance Eq Foo where
    (==) = undefined
