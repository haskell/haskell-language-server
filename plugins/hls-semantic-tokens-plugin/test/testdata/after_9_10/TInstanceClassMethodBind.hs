module TInstanceClassMethodBind where


data Foo = Foo Int
instance Show Foo where
    show = undefined
