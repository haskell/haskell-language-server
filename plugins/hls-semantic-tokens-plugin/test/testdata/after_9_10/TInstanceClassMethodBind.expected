[1:1-7 HsSyntacticTokenType TKeyword "module"]
[1:8-32 HsSemanticTokenType TModule "TInstanceClassMethodBind"]
[1:33-38 HsSyntacticTokenType TKeyword "where"]
[4:1-5 HsSyntacticTokenType TKeyword "data"]
[4:6-9 HsSemanticTokenType TTypeConstructor "Foo"]
[4:10-11 HsSyntacticTokenType TKeyword "="]
[4:12-15 HsSemanticTokenType TDataConstructor "Foo"]
[4:16-19 HsSemanticTokenType TTypeConstructor "Int"]
[5:1-9 HsSyntacticTokenType TKeyword "instance"]
[5:10-14 HsSemanticTokenType TClass "Show"]
[5:15-18 HsSemanticTokenType TTypeConstructor "Foo"]
[5:19-24 HsSyntacticTokenType TKeyword "where"]
[6:5-9 HsSemanticTokenType TClassMethod "show"]
[6:10-11 HsSyntacticTokenType TKeyword "="]
[6:12-21 HsSemanticTokenType TVariable "undefined"]
